src/main/scala/org/broadinstitute/dsde/workbench/sam/azure/AzureRoutes.scala

```diff
@@ -71,7 +71,19 @@ trait AzureRoutes extends SecurityDirectives with LazyLogging {
                     }
                   }
                 }
-              }
+              } ~
+                delete {
+                  requireAction(
+                    FullyQualifiedResourceId(SamResourceTypes.spendProfile, ResourceId(billingProfileId)),
+                    SamResourceActions.delete,
```

Comment:
I am not super familiar with how our auth middleware is setup here, just to double check is there any additional authorization we can/should check on delete? On create it looks like there is a `requireCloudExtensionCreatePetAction` maybe we need a similar deletion requirement?